src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -32,122 +45,200 @@ class ScatterPlot extends Visualization {
+    addZooming(box_width, box_height, points, scatter, x, xMin, padding_x, xMax, xAxis, y) {
+        let brush = d3.brushX()
+            .extent([[0, 0], [box_width, box_height]])
+            .on("end", updateChart)
 
-            scatter
-                .selectAll("circle")
-                .data(_data)
-                .enter()
-                .append("circle")
-                .attr("cx", function (d) {
-                    return x(d[headerNames[0]]);
-                })
-                .attr("cy", function (d) {
-                    return y(d[headerNames[1]]);
-                })
-                .attr("r", 8)
-                .style("fill", function (d) {
-                    return color(d[headerNames[2]])
-                })
-                .style("opacity", 0.5)
+        scatter
+            .append("g")
+            .attr("class", "brush")
+            .call(brush);
 
-            scatter
-                .append("g")
-                .attr("class", "brush")
-                .call(brush);
+        let idleTimeout
 
-            var idleTimeout
+        function idled() {
+            idleTimeout = null;
+        }
 
-            function idled() {
-                idleTimeout = null;
-            }
+        function updateChart() {
+            let extent = d3.event.selection;
 
-            function updateChart() {
-                let extent = d3.event.selection;
+            if (!extent) {
+                if (!idleTimeout) return idleTimeout = setTimeout(idled, 350);
+                x.domain([xMin - padding_x, xMax + padding_x]);
+            } else {
+                x.domain([x.invert(extent[0]), x.invert(extent[1])]);
+                scatter.select(".brush").call(brush.move, null);
+            }
 
-                if (!extent) {
-                    if (!idleTimeout) return idleTimeout = setTimeout(idled, 350);
-                    x.domain([4, 8])
-                } else {
-                    x.domain([x.invert(extent[0]), x.invert(extent[1])])
-                    scatter.select(".brush").call(brush.move, null)
-                }
+            xAxis.transition().duration(1000).call(d3.axisBottom(x));
+            scatter
+                .selectAll("path")
+                .transition().duration(1000)
+                .attr('transform', d => "translate(" + x(d.x) + "," + y(d.y) + ")")
 
-                xAxis.transition().duration(1000).call(d3.axisBottom(x))
+            if (points.labels === "visible") {
                 scatter
-                    .selectAll("circle")
+                    .selectAll("text")
                     .transition().duration(1000)
-                    .attr("cx", function (d) {
-                        return x(d[headerNames[0]]);
-                    })
-                    .attr("cy", function (d) {
-                        return y(d[headerNames[1]]);
-                    })
-
+                    .attr('transform', d => "translate(" + x(d.x) + "," + y(d.y) + ")")
```

Comment:
The `transition().duration(...).attr(...) pattern seems repetitive, maybe make some lambda from it?

Reply:
removed as other zooming method will be used.